### ISSUE TITLE
SIGSEGV caused by processors

### ISSUE THREAD

**Issue by mjf** [influxdata/telegraf#8089]
### Relevant telegraf.conf:
```toml
[agent]
  hostname = "test.example.com"
[[outputs.influxdb_v2]]
  bucket = "test"
  organization = "Test Organization"
  token = "REDACTED"
  urls = ["https://test.example.com:9999"]
[[inputs.http_listener_v2]]
    name_override = "test"
    service_address = "test.example.com:8001"
    methods = [ "GET" ]
    path = "/test"
    data_source = "query"
    data_format = "form_urlencoded"
    form_urlencoded_tag_keys = ["chan", "unit", "name", "id", "val", "min", "max"]
    # fielddrop = ["val_1"]
[[processors.regex]]
    order = 1
    namepass = ["test"]
    [[processors.regex.tags]]
      key = "val"
      pattern = "^(%20)+"
      replacement = ""
    [[processors.regex.tags]]
      key = "min"
      pattern = "^(%20)+"
      replacement = ""
    [[processors.regex.tags]]
      key = "max"
      pattern = "^(%20)+"
      replacement = ""
[[processors.converter]]
    order = 2
    namepass = ["test"]
    [processors.converter.tags]
      float = ["val", "min", "max"]
```

### System info:
``Telegraf 1.15.2 (git: HEAD cd037b49)``

### Docker

No *Docker*...

### Steps to reproduce:
With the provided configuration *Telegraf* crashes on the following ``GET`` request:

```shell
$ curl 'http://test.example.com:8001/test?chan=01&name=measurement%20test&stat=0&val=%20%20%201.234&unit=Pa&max=0&min=0&mode=0&id=test'
```

1. start *Telegraf* with configuration similar to the provided sample configuration
2. submit the provided sample ``GET`` request

### Expected behavior:

*Telegraf* operates as expected and does not crash with ``SIGSEGV``.

### Actual behavior:
*Telegraf* crashes with ``SIGSEGV``:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x1f92c58]

goroutine 23 [running]:
github.com/influxdata/telegraf/plugins/processors/converter.(*Converter).convertTags(0xc000029f80, 0x2f59ae0, 0xc0004b2000)
        /go/src/github.com/influxdata/telegraf/plugins/processors/converter/converter.go:217 +0x358
github.com/influxdata/telegraf/plugins/processors/converter.(*Converter).Apply(0xc000029f80, 0xc000060c90, 0x1, 0x1, 0xc00006de40, 0xc00006ded8, 0x7e03ba)
        /go/src/github.com/influxdata/telegraf/plugins/processors/converter/converter.go:85 +0x7a
github.com/influxdata/telegraf/plugins/processors.(*streamingProcessor).Add(0xc00042d4c0, 0x2f59ae0, 0xc0004b2000, 0x2f4aa80, 0xc00000e820, 0x1, 0xc000000101)
        /go/src/github.com/influxdata/telegraf/plugins/processors/streamingprocessor.go:35 +0x81
github.com/influxdata/telegraf/models.(*RunningProcessor).Add(0xc00055c150, 0x2f59ae0, 0xc0004b2000, 0x2f4aa80, 0xc00000e820, 0x2, 0x1)
        /go/src/github.com/influxdata/telegraf/models/running_processor.go:95 +0x106
github.com/influxdata/telegraf/agent.(*Agent).runProcessors.func1(0xc0004fa000, 0xc00000e620)
        /go/src/github.com/influxdata/telegraf/agent/agent.go:559 +0x156
created by github.com/influxdata/telegraf/agent.(*Agent).runProcessors
        /go/src/github.com/influxdata/telegraf/agent/agent.go:554 +0x90
```

### Additional info:
I found the bug while trying to workaround issues described in #8064.

**Comment by ssoroka** [comment]
Can you supply the input metrics? It looks like it's having trouble converting a value to a float then crashing on error logging?

**Comment by mjf** [comment]
@ssoroka Hello. Thanks for you attention. They are there (sort of). Please, see the `curl` query in the report...

The point of the whole thing is to try to use processors to fix the *URL* variables before it even gets parsed by the `form_urlencoded`. *Telegraf* lacks "preprocessors" at input (from my point of view some `preprocessors.regex` would
be sufficient in my specific case described more further in the reference PR, but there's no such a thing yet). So the idea behind it is this (it's a bit like *scratching with ones left foot on ones right ear*): :smile: 

1. get all the variables as tags (and hope I get strings),
2. fix all ill-formed values in tags by `processors.regex`,
3. convert the tags with the fixed values back to fields by using `processors.converter`,
4. finaly pass it to the `processor.form_urlencoded` parser,
5. send metrices to a consumer (i.e. *InfluxDB*).

I'm lacking more ideas how to get my metrices from the buggy devices.,. :disappointed: Either I use what is already provided by *Telegraf* or I need to patch it (but I can not and do not want to use some custom build in production, especially not due to such a minor and uninfluenceble change as shown in the PR).

**Comment by ssoroka** [comment]
> Please, see the curl query in the report

test.example.com is not queryable.

Your general idea is sound. You will need to specify the `order` to make sure they flow in the order you expect (it's random otherwise), and you might also be interested in the starlark and execd processors if you get stuck, which allow total freedom in manipulating the data with custom scripts/programs.

Let me know if you want any more information on these.

**Comment by mjf** [comment]
@ssoroka 

> > Please, see the curl query in the report
> 
> test.example.com is not queryable.

Of course ([RFC6771](https://tools.ietf.org/html/rfc6761#section-6.5)). What shall I put in the issue next time?

I wrote the bug report in a way so that anybody could use it to configure *Telegraf*, *InfluxDB* and see exactly the same thing.
It's common to redact all sensitive information and use i.e. `example.com` for domain name etc., as far as I know. It's upon the report receiver to cope with it. I did my best to allow you to replicate the crash. That's all I can do about it. *Telegraf* is crashing! Isnt't that a really bad thing that shouldn't happen in production source code?

**Comment by zorel** [comment]
I have almost the same issue with converter plugin, when trying to convert this:
```
Timestamp,"VENGEANCE RGB PRO Temp. #1","VENGEANCE RGB PRO Temp. #2","COMMANDER PRO Temp. GPU","COMMANDER PRO Temp. CPU","COMMANDER PRO Temp.  Alimentation","COMMANDER PRO 12V","COMMANDER PRO 5V","COMMANDER PRO 3.3V","H100i RGB PRO XT Temp.","H100i RGB PRO XT Ventilateur #1","H100i RGB PRO XT Ventilateur #2","H100i RGB PRO XT Pompe"
15/9/2020 07:27:18,30.00°C,30.00°C,28.81°C,27.92°C,27.89°C,11.95V,4.91V,3.37V,22.70°C,816RPM,790RPM,2349RPM
```
 with this configuration:
```
[[processors.converter]]
  order = 2
  [processors.converter.fields]
    float = ["*"]
[[inputs.file]]
  files = ["path_to_file.csv"]
  data_format = "csv"
  csv_header_row_count = 1
  csv_timestamp_column = "Timestamp"
  csv_timestamp_format = "2/1/2006 15:04:05"

```

Error occurs on converter.go line 265:
`p.Log.Errorf("error converting to float [%T]: %v", value, value)`

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x20f4425]

goroutine 50 [running]:
github.com/influxdata/telegraf/plugins/processors/converter.(*Converter).convertFields(0xc0001d5c80, 0x2f3e980, 0xc0001da000)
        /home/zorel/telegraf/plugins/processors/converter/converter.go:265 +0xa85
github.com/influxdata/telegraf/plugins/processors/converter.(*Converter).Apply(0xc0001d5c80, 0xc001cc3420, 0x1, 0x1, 0xc000200a80, 0xc000200840, 0xc000200838)
        /home/zorel/telegraf/plugins/processors/converter/converter.go:86 +0x9c
github.com/influxdata/telegraf/plugins/processors.(*streamingProcessor).Add(0xc000568240, 0x2f3e980, 0xc0001da000, 0x2f33860, 0xc0005688c0, 0xc000732701, 0xc000560101)
        /home/zorel/telegraf/plugins/processors/streamingprocessor.go:35 +0x82
github.com/influxdata/telegraf/models.(*RunningProcessor).Add(0xc0001d5e00, 0x2f3e980, 0xc0001da000, 0x2f33860, 0xc0005688c0, 0xc000111f00, 0x2540be400)
        /home/zorel/telegraf/models/running_processor.go:95 +0x108
github.com/influxdata/telegraf/agent.(*Agent).runProcessors.func1(0xc0001c73b0, 0xc000568820)
        /home/zorel/telegraf/agent/agent.go:559 +0x157
created by github.com/influxdata/telegraf/agent.(*Agent).runProcessors
        /home/zorel/telegraf/agent/agent.go:554 +0x91
```

**Comment by a-bali** [comment]
I am also getting this, SIGSEGV apparently occurring on any kind of logging calls (both debug and error) in a processor plugin.

**Comment by ssoroka** [comment]
Ok, the issue was that the logger wasn't being initialized properly. These are all fixed in v1.16.0-rc1, which is out now. Sorry about the confusion.
